src/app/search/search.component.ts

```diff
@@ -499,24 +500,28 @@ export class SearchComponent implements OnInit, OnDestroy {
    */
   updateResultsTable(value: string) {
     this.alertService.start('Performing search request');
-    this.extendedGA4GHService.toolsIndexSearch(value).subscribe(
-      (hits: any) => {
-        this.hits = hits.hits.hits;
-        const filteredHits: [Array<Hit>, Array<Hit>, Array<Hit>] = this.searchService.filterEntry(this.hits, this.query_size);
-        const searchText = this.searchQuery.getValue().searchText;
-        this.searchService.setHits(filteredHits[0], filteredHits[1], filteredHits[2]);
-        if (searchText.length > 0 && hits) {
-          this.searchTerm = true;
-        }
-        if (this.searchTerm && this.hits.length === 0) {
-          this.searchService.suggestSearchTerm(searchText);
+    this.isLoading = true;
+    this.extendedGA4GHService
+      .toolsIndexSearch(value)
+      .pipe(finalize(() => (this.isLoading = false)))
```

Comment:
This and line 503 are the changes here; the rest of the changes in the block are just formatting.